corehq/apps/users/migrations/0072_remove_location_fields_in_user_data.py

```diff
@@ -0,0 +1,63 @@
+from django.db import migrations
+
+from couchdbkit import ResourceNotFound
+
+from corehq.apps.users.util import user_location_data
+from corehq.apps.users.models import CommCareUser
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+@skip_on_fresh_install
+def remove_keys_from_data(apps, schema_editor):
+    SQLUserData = apps.get_model('users', 'SQLUserData')
+
+    for user_data in SQLUserData.objects.all():
```

Comment:
You might want to change this to make it so you can run the migration before deploy.  I think there are two options:
1. Add a filter to this queryset to include only rows that have one or more of the keys present (`user_data.data` is a `JSONField`, and so should be queryable)
2. Add a `@once_off_migration` decorator somewhere.

Reply:
Hi Ethan, why I need to change it to run the migration before deploy? I plan to run this after the PR to read from system data is merged.

Reply:
We usually try to run migrations before the deploy if it might take a while, so as not to complicate the deploy itself.  Are you planning to deploy a private release and run the migration from there?  That could also work.

Reply:
That should be fine then.  Sometimes it's preferred to run the actual data migration in a management command, and then have a django migration that calls that command to ensure it gets run on deploy later.  For that approach to work, you need something to either skip the already-migrated rows or skip the whole migration.  However, if you'll be running it via the django migration, then the migration framework essentially does that anyways, so no special handling is required here.

Reply:
Thank you for your explanation! I will run migration before I merge the PR, so I can avoid blocking production deployment.

Reply:
Actually I think it's a best practice to merge before running the migration, that way all migrations on production are at least on `master`.  This is also a failsafe to ensure no one else merges a conflicting migration in the interim, which would be a mess to resolve.

There's some applicable information in this doc if helpful: https://commcare-hq.readthedocs.io/migrations_in_practice.html#
(Though I've been meaning to revisit/rewrite it)
And this doc describes the migration command pattern: https://commcare-hq.readthedocs.io/migration_command_pattern.html